plutarch-test/src/Plutarch/Test.hs

```diff
@@ -0,0 +1,205 @@
+import System.FilePath
+import Test.Syd (
+  Expectation,
+  Spec,
+  TestDefM,
+  describe,
+  expectationFailure,
+  getTestDescriptionPath,
+  it,
+  pureGoldenTextFile,
+  shouldBe,
+ )
+
+import Plutarch
+import Plutarch.Benchmark (benchmarkScript')
+import Plutarch.Bool
+import Plutarch.Evaluate
+import qualified Plutus.V1.Ledger.Scripts as Scripts
+
+{- |
+    Like `shouldBe` but but for Plutarch terms
+-}
+pshouldBe :: forall (a :: PType) (b :: PType). ClosedTerm a -> ClosedTerm b -> Expectation
+pshouldBe x y = do
+  p1 <- fmap printScript $ eval $ compile x
+  p2 <- fmap printScript $ eval $ compile y
+  p1 `shouldBe` p2
+  where
+    eval :: Scripts.Script -> IO Scripts.Script
+    eval s = case evaluateScript s of
+      Left e -> expectationFailure $ "Script evaluation failed: " <> show e
+      Right (_, _, x') -> pure x'
+
+{- Like `@?=` but for Plutarch terms -}
+(#@?=) :: forall (a :: PType) (b :: PType). ClosedTerm a -> ClosedTerm b -> Expectation
+(#@?=) = pshouldBe
+
+{- Asserts the term to be true -}
+passert :: forall (a :: PType). ClosedTerm a -> Expectation
+passert p = p #@?= pcon PTrue
+
+{- Asserts the term evaluates successfully without failing -}
+psucceeds :: forall (a :: PType). ClosedTerm a -> Expectation
+psucceeds p =
+  case evaluateScript (compile p) of
+    Left _ -> expectationFailure $ "Term failed to evaluate"
+    Right _ -> pure ()
+
+{- Like `printTerm` but evaluates the term beforehand.
+
+  All evaluation failures are treated as equivalent to a `perror`. Plutus does
+  not provide an accurate way to tell if the program evalutes to `Error` or not;
+  see https://github.com/input-output-hk/plutus/issues/4270
+
+-}
+printTermEvaluated :: forall a. ClosedTerm a -> String
+printTermEvaluated p =
+  case evaluateScript (compile p) of
+    Left _ -> printTerm perror
+    Right (_, _, x) -> printScript x
+
+{- | Asserts that the term evaluates successfully with the given trace sequence
+
+  See also: `plutarchDevFlagDescribe`
+-}
+ptraces :: forall (a :: PType). ClosedTerm a -> [Text] -> Expectation
+ptraces p develTraces =
+  case evaluateScript (compile p) of
+    Left _ -> expectationFailure $ "Term failed to evaluate"
+    Right (_, traceLog, _) -> do
+#ifdef Development 
+      traceLog `shouldBe` develTraces
+#else
+      -- Tracing is disabled in non-developed modes, so we should expect an
+      -- empty trace log.
+      let noTraces = const [] develTraces
+      traceLog `shouldBe` noTraces
+#endif
+
+{- | Like `describe`, but determines description from `Development` CPP flag
+
+  Useful to create two sets of othersise identical group of tests that differ
+  only by `Development` flag. This has the effect of creating two sets of golden
+  tests (with different filepaths) for corresponding flag values.
+
+  Typically meant to be used in conjunction with `ptraces`.
+-}
+plutarchDevFlagDescribe :: forall (outers :: [Type]) inner. TestDefM outers inner () -> TestDefM outers inner ()
```

Comment:
Ah, you use this to choose which to test twice?

Reply:
Yup, the idea is to use `plutarchDevFlagDescribe` somewhere in test hierarchy where you know that the child sub-tree of tests also use Plutarch programs that will behave differently across 'development' flag sets (eg: with tracing). I'll add this to the README.